From 0e222876d7da7acbaaf955c2e7b7efdf3549bde9 Mon Sep 17 00:00:00 2001
From: Chase Douglas <chase.douglas@canonical.com>
Date: Fri, 08 Jun 2012 20:53:38 +0000
Subject: Turn down the default sensitivity for unibody MacBook touchpads

The touchpads are too sensitive by default. A touch is activated when a
finger hovers a millimeter off the device. This change turns down the
default sensitivity so that accidental touchpad interactions are
lessened and pointer interaction accuracy is increased.

Signed-off-by: Chase Douglas <chase.douglas@canonical.com>
Reviewed-by: Peter Hutterer <peter.hutterer@who-t.net>
Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>
---
diff --git a/src/synaptics.c b/src/synaptics.c
index 68d35e8..2144684 100644
--- a/src/synaptics.c
+++ b/src/synaptics.c
@@ -319,17 +319,23 @@ static void
 calculate_tap_hysteresis(SynapticsPrivate * priv, int range,
                          int *fingerLow, int *fingerHigh)
 {
-    if (priv->model == MODEL_ELANTECH) {
+    switch (priv->model) {
+    case MODEL_ELANTECH:
         /* All Elantech touchpads don't need the Z filtering to get the
          * number of fingers correctly. See Documentation/elantech.txt
          * in the kernel.
          */
         *fingerLow = priv->minp + 1;
         *fingerHigh = priv->minp + 1;
-    }
-    else {
+        break;
+    case MODEL_UNIBODY_MACBOOK:
+        *fingerLow = 70;
+        *fingerHigh = 75;
+        break;
+    default:
         *fingerLow = priv->minp + range * (25.0 / 256);
         *fingerHigh = priv->minp + range * (30.0 / 256);
+        break;
     }
 }
 
--
cgit v0.9.0.2-2-gbebe
